vnext/Desktop/DevSupportManager.cpp

```diff
@@ -55,60 +55,49 @@ string DevSupportManager::GetJavaScriptFromServer(
+  client.DefaultRequestHeaders().Host(HostName(uri.Host().c_str()));
+
+  HttpRequestMessage request(HttpMethod::Get(), uri);
+  HttpResponseMessage response = client.SendRequestAsync(request).get();
+
+  IBuffer buffer = response.Content().ReadAsBufferAsync().get();
+  auto reader = DataReader::FromBuffer(buffer);
+  reader.UnicodeEncoding(UnicodeEncoding::Utf8);
+  auto length = reader.UnconsumedBufferLength();
+  string result;
+  if (length > 0 || response.IsSuccessStatusCode()) {
+    vector<uint8_t> data(length);
+    reader.ReadBytes(data);
+    result = string(reinterpret_cast<char *>(data.data()), data.size());
+  } else {
     m_exceptionCaught = true;
-    return R"({"error:")"s + ec.message() + R"("})"s;
+    ostringstream stream;
+    stream << R"({"error:")"s << static_cast<int>(response.StatusCode()) << " downloading " << bundleUrl << R"("})"s;
+    result = stream.str();
   }
 
-  request<string_body> request{verb::get, url.Target(), 11};
-  request.set(field::host, url.host);
-  request.set(field::user_agent, BOOST_BEAST_VERSION_STRING);
-
-  write(socket, request);
-
-  flat_buffer buffer;
-  response<string_body> response;
-
-  parser<false, string_body> p{std::move(response)};
-  p.body_limit(25 * 1024 * 1024); // 25MB (boost default of 1MB is too small for dev bundles)
-
-  read(socket, buffer, p.base());
-  response = p.release();
-  std::stringstream jsStringStream;
-  jsStringStream << response.body();
-  // TODO: Check if UTF-8 processing is required.
-
-  return jsStringStream.str();
+  return result;
 }
 
 void DevSupportManager::StartPollingLiveReload(const string &debugHost, std::function<void()> onChangeCallback) {
   auto t = create_task([this, debugHost, onChangeCallback = move(onChangeCallback)] {
     cancellation_token token = m_liveReloadCts.get_token();
     while (!token.is_canceled()) {
       try {
-        Url url(DevServerHelper::get_OnChangeEndpointUrl(debugHost));
-        auto const resolveResult = m_resolver.resolve(url.host, url.port);
-        tcp::socket socket{m_context};
-        connect(socket, resolveResult);
-
-        request<string_body> request{verb::get, url.Target(), 11 /*HTTP 1.1*/};
-        request.set(field::host, url.host);
-        request.set(field::user_agent, BOOST_BEAST_VERSION_STRING);
-        request.set(field::connection, "keep-alive");
-
-        write(socket, request);
-        flat_buffer buffer;
-        response<dynamic_body> response;
-        read(socket, buffer, response);
-
-        if (response.result_int() == 205 /*ResetContent*/ && !token.is_canceled())
+        Uri uri(Common::Unicode::Utf8ToUtf16(DevServerHelper::get_OnChangeEndpointUrl(debugHost)));
+        HttpClient client;
+        client.DefaultRequestHeaders().Host(HostName(uri.Host().c_str()));
+        client.DefaultRequestHeaders().Connection().TryParseAdd(L"keep-alive");
+
+        HttpResponseMessage response = client.GetAsync(uri, HttpCompletionOption::ResponseHeadersRead).get();
```

Comment:
>.get(); [](start = 102, length = 7)

I think we could do better here without blocking the thread. We can simply add a continuation for the action returned here which will be executed in a thread pool and decides what to do next: repeat or exit.
We can probably do it using the co_await.

Reply:
Completely agreed, but this is out of the scope of this change.
This needs to be backported to `0.60-stable` to resolve an Office bug related to Boost.

Once deployed, I'll send another PR addressing thread blocking.